src/services/state-persistor/utils/FeatureHash.ts

```diff
@@ -0,0 +1,1399 @@
+import ngeoFormatFeatureProperties from './FeatureProperties'
+import { rgbArrayToHex } from './utils'
+
+import olFeature from 'ol/Feature.js'
+import olMap from 'ol/Map.js'
+import { includes as arrayIncludes } from 'ol/array.js'
+import { asArray as colorAsArray } from 'ol/color.js'
+import { ReadOptions, transformGeometryWithOptions } from 'ol/format/Feature.js'
+import olGeomGeometryLayout from 'ol/geom/GeometryLayout.js'
+import olGeomLineString from 'ol/geom/LineString.js'
+import olGeomMultiLineString from 'ol/geom/MultiLineString.js'
+import olGeomMultiPoint from 'ol/geom/MultiPoint.js'
+import olGeomMultiPolygon from 'ol/geom/MultiPolygon.js'
+import olGeomPoint from 'ol/geom/Point.js'
+import olGeomPolygon from 'ol/geom/Polygon.js'
+import olStyleCircle from 'ol/style/Circle.js'
+import olStyleFill from 'ol/style/Fill.js'
+import olStyleStroke from 'ol/style/Stroke.js'
+import olStyleStyle from 'ol/style/Style.js'
+import olStyleText from 'ol/style/Text.js'
+import Feature from 'ol/Feature.js'
+import {
+  Geometry,
+  LineString,
+  MultiLineString,
+  MultiPoint,
+  MultiPolygon,
+  Point,
+  Polygon,
+} from 'ol/geom'
+import TextFeature from 'ol/format/TextFeature.js'
+import Text from 'ol/style/Text.js'
+import Stroke from 'ol/style/Stroke.js'
+import Fill from 'ol/style/Fill.js'
+import Style from 'ol/style/Style.js'
+import { createStyleFunction } from '@/composables/draw/draw-utils'
+
+const GeometryTypeValues = {
+  LineString: 'LineString',
+  Point: 'Point',
+  Polygon: 'Polygon',
+  MultiLineString: 'MultiLineString',
+  MultiPoint: 'MultiPoint',
+  MultiPolygon: 'MultiPolygon',
+}
+
+type GeometryType = keyof typeof GeometryTypeValues
+
+/**
+ * @enum {string}
+ */
+const FeatureHashStyleType = {
+  LINE_STRING: 'LineString',
+  POINT: 'Point',
+  POLYGON: 'Polygon',
+}
+
+/**
+ * @type {Object.<ol.geom.GeometryType, FeatureHashStyleType>}
+ * @private
+ */
+
+const FeatureHashStyleTypes = {
+  LineString: FeatureHashStyleType.LINE_STRING,
+  Point: FeatureHashStyleType.POINT,
+  Polygon: FeatureHashStyleType.POLYGON,
+  MultiLineString: FeatureHashStyleType.LINE_STRING,
+  MultiPoint: FeatureHashStyleType.POINT,
+  MultiPolygon: FeatureHashStyleType.POLYGON,
+}
+
+/**
+ * @type {Object.<string, string>}
+ * @private
+ */
+let FeatureHashLegacyProperties: any = {}
```

Comment:
can't we provide a better typing than any? what about the jsdoc above saying <string, string>?

Reply:
for the moment, I made a linter exception because it was a big piece of code to port to TS